dcpy/test/utils/test_s3.py

```diff
@@ -8,9 +9,38 @@
 TEST_FILE_NAME = "test-file"
 
 
+def test_mock_credentials(aws_credentials):
+    s3.client()
+
+
+# assert expected results for invalid client endpoint_url options
+def test_client_create_errors(aws_credentials):
+    with pytest.raises(TypeError, match="str expected, not NoneType"):
+        os.environ["AWS_S3_ENDPOINT"] = None
+
+    with pytest.raises(ValueError, match="Invalid endpoint: testing"):
+        os.environ["AWS_S3_ENDPOINT"] = "testing"
+        s3.client()
+
+    with pytest.raises(ValueError, match="Invalid endpoint: "):
+        os.environ["AWS_S3_ENDPOINT"] = ""
+        s3.client()
+
+
+@pytest.mark.skip(reason="takes ~5 seconds")
```

Comment:
Hmm... maybe you can set the connection timeout for s3 to be a very small unit of time?

Reply:
i _think_ @fvankrieken set the timeout to 120 seconds [here](https://github.com/NYCPlanning/data-engineering/pull/94/commits/7cdf852835ef6fc4e6971f04b1bb4fd7dda1c2b7) because some files really took longer than the default 60 seconds

this test was mainly just to help me figure out where boto was getting mad about various endpoint url formats. happy to drop it. if someone sets a bad url, boto will raise this descriptive error

Reply:
Yeah, I was thinking that in tests you could potentially override the timeout. Happy with whatever route you go though